File: AmsterdamEvents/src/app/models/a-event.ts
import {AEventStatus} from "./a-event-status";

export class AEvent {
  title: String = "Amazing event";
  description: String = "just come";
  start: Date;
  end: Date;
  status: AEventStatus | string;
  isTicketed: boolean;
  participationFee: number;
  maxParticipants: number;
  id: number = AEvent.beginId;
  static beginId: number = 19999;
  static getNextId = () => ++AEvent.beginId;

  constructor(id: number, title: String, start: Date, end: Date, description: String, status: AEventStatus | string, isTicketed: boolean,
              participationFee: number, maxParticipants: number) {
    this.title = title;
    this.start = start;
    this.end = end;
    this.description = description;
    this.status = status;
    this.isTicketed = isTicketed;
    this.participationFee = participationFee;
    this.maxParticipants = maxParticipants;
  }

  public static createRandomAEvent(): AEvent {
    let title: String = "Amazing event";
    let description: String = "just come";

    return new AEvent(this.getNextId(), title, this.randomStart(), this.randomDuration(), description, this.randomStatus(),
      this.randomIsTicketed(), this.randomFee(), this.randomParticipants()
    )
  }

  public static randomStatus(): string {
    let key = Math.floor(Math.random() * (Object.keys(AEventStatus).length+1) / 2);
    while (key > 2) {
       key = Math.floor(Math.random() * (Object.keys(AEventStatus).length+1) / 2);
    }
    return AEventStatus[key];
  }

  public static randomStart(): Date {
    let startDate = Date.now() + Math.random() * 100000000000;

    return new Date(startDate);
  }

  public static randomDuration(): Date {
    const startDate = Date.now() + Math.random() * 100000000000;
    const endDate = startDate + Math.random() * 10000000;
    const duration = (endDate - startDate);
    let timestamp = Math.round(Math.random() * duration);
    timestamp += startDate;

    return new Date(timestamp);
  }

  public static randomIsTicketed(): boolean {
    return Math.random() < 0.5;
  }

  public static randomFee(): any {
    if (this.randomIsTicketed()) {
      return (Math.random() * (30 - 1 + 1) + 1).toFixed(2);
    } else return "free"
  }

  public static randomParticipants(): any {
    if (this.randomIsTicketed()) {
      return Math.round(Math.random() * (10 - 1 + 1) + 1) + "00";
    }
  }
}

File: aeserver/src/main/java/com/example/aeserver/models/AEvent.java
package com.example.aeserver.models;


import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import java.io.Serializable;
import java.util.*;

/**
 *
 *
 * @author Nilava Kazal [studentennummer: 500847707] 07/11/2021 13:55
 */

@Entity(name = "AEvent")
public class AEvent implements Serializable {
    String title = "Amazing event";

    @Column(name = "description")
    String description = "just come";

    @Column(name = "start_date")
    String start;

    @Column(name = "end_date")
    String end;

    @Column(name = "status")
    String status;

    @Column(name = "is_ticketed")
    boolean isTicketed;

    @Column(name = "participation_fee")
    double participationFee;

    @Column(name = "max_participants")
    String maxParticipants;

    int id;
    static int beginId = 19999;

    @Id
    @Column(name = "id")
    static int getNextId = ++AEvent.beginId;

    public AEvent(String title) {
        this.id = getNextId;
        this.title = title;
        this.start = randomDate();
        this.end = randomDate();
        this.description = description;
        this.status = randomStatus();
        this.isTicketed = randomIsTicketed();
        this.participationFee = randomFee();
        this.maxParticipants = randomParticipants();
    }


    public static String randomStatus() {
        List<String> statuses = Arrays.asList("DRAFT", "PUBLISHED", "CANCELED");
        Random rand = new Random();

        return statuses.get(rand.nextInt(statuses.size()));
    }

    public static String randomDate() {
        Random rnd = new Random();
        Date date = new Date(Math.abs(System.currentTimeMillis() - rnd.nextLong()));
        return date.toString();
    }

//    public static int randomDuration() {
//        int start = randomDate();
//        int end = (int) (randomDate() + Math.random() * 10000000);
//
//        return start + (int) Math.round(Math.random() * (end - start));
//    }

    public static boolean randomIsTicketed() {
        return Math.random() < 0.5;
    }

    public static double randomFee() {
        if (randomIsTicketed()) {
            return (Math.random() * (30 - 1 + 1) + 1);
        } else return 00.0;
    }


    public static String randomParticipants() {
        if (randomIsTicketed()) {
            return Math.round(Math.random() * (10 - 1 + 1) + 1) + "00";
        } else return null;
    }

    public static AEvent createRandomAEvent() {
        String title = "Amazing event";
        String description = "just come";

        return new AEvent(title);
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getStart() {
        return start;
    }

    public String getEnd() {
        return end;
    }

    public String getStatus() {
        return status;
    }

    public boolean isTicketed() {
        return isTicketed;
    }

    public double getParticipationFee() {
        return participationFee;
    }

    public String getMaxParticipants() {
        return maxParticipants;
    }

    public int getId() {
        return id;
    }

    public static int getBeginId() {
        return beginId;
    }

    public static int getGetNextId() {
        return getNextId;
    }
}
